include/revng-c/RestructureCFG/CycleEquivalenceAnalysis.h

```diff
@@ -0,0 +1,257 @@
+#pragma once
+
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include <map>
+
+#include "llvm/IR/CFG.h"
+#include "llvm/Pass.h"
+
+#include "revng/ADT/GenericGraph.h"
+#include "revng/Support/Assert.h"
+
+#include "revng-c/ADT/STLExtras.h"
+
+#include "CycleEquivalenceClass.h"
+#include "CycleEquivalenceResult.h"
+
+enum class OriginalEdgeKind {
+  Real,
+  Fake,
+  Invalid,
+};
+
+enum class SpanningTreeEdgeKind {
+  TreeEdge,
+  BackEdge,
+  Invalid,
+};
+
+/// The `EdgeLabel` `struct` is used to represent the information needed to
+/// identify each undirected edge in the `GenericGraph` materialized internally
+/// in the analysis, and to map it eventually on the edge on the undirected
```

Comment:
I think you meant directed here, not undirected.

Reply:
Right, it was a typo, corrected it [here](https://github.com/revng/revng-c/blob/f23593da9f15288aca2fee41975e6e55dbbbcb3b/include/revng-c/RestructureCFG/CycleEquivalenceAnalysis.h#L34)